src/adapters/scene_direction.js

```diff
@@ -122,6 +122,11 @@ export default class SceneDirection {
       this.map.getSource(`source_${route.id}`).setData(this.buildRouteData([]))
```

Comment:
I think these empty coordinates `[]` may cause weird behavior of mapbox-gl-js ?  
Why not simply use `removeLayer` here ?

Reply:
You are right, i'll remove the source (now i know it's possible) then remove the "update data source" case on set route